django_declarative_apis/machinery/filtering.py

```diff
@@ -67,6 +68,36 @@ def _get_unexpanded_field_value(inst, field_name, field_type):
         return {display_key: getattr(obj, display_key)}
 
 
+def _is_relation(field):
+    return field.is_relation or getattr(field, "is_fake_relation", False)
+
+
+def _is_reverse_relation(field):
+    return isinstance(field, ForeignObjectRel)
+
+
+def _cache_related_instance(inst, field_name, model_cache):
+    try:
+        field_meta = inst._meta.get_field(field_name)
+    except (AttributeError, FieldDoesNotExist):
+        return
+    if not _is_relation(field_meta) or _is_reverse_relation(field_meta):
+        return  # no `attname` in reverse relations
+    val_pk = getattr(inst, field_meta.attname)
+    val_cls = field_meta.related_model
+    cache_key = _make_model_cache_key(val_cls, val_pk)
+    if cache_key in model_cache:
+        logger.debug("ev=model_cache, status=hit, key=%s", cache_key)
+        related_instance = model_cache[cache_key]
+    else:
+        logger.debug("ev=model_cache, status=miss, key=%s", cache_key)
+        related_instance = getattr(inst, field_name)
+        if not isinstance(related_instance, models.Model):
```

Comment:
Seems very generic, can't we use the instance type of `val_cls`

Reply:
good catch. yeah, I like it

Reply:
added it